doc/source/rllib/connector-v2.rst

```diff
@@ -0,0 +1,596 @@
+            super().__init__(env)
+            self.min = min_
+            self.max = max_
+        # Override `reward` to custom process the original reward coming
+        # from the env.
+        def reward(self, reward):
+            # E.g. simple clipping between min and max.
+            return np.clip(reward, self.min, self.max)
+
+The simplest way of customizing an env-to-module pipeline is to write an
+:py:class:`~ray.rllib.connectors.env_to_module.observation_preprocessor.ObservationPreprocessor` and plug
+it into the :py:class:`~ray.rllib.algorithms.algorithm_config.AlgorithmConfig`. All you have to do in this case is implement
+2 methods:
+
+.. testcode::
+
+    import gymnasium as gym
+    import numpy as np
+
+    from ray.rllib.connectors.env_to_module.observation_preprocessor import ObservationPreprocessor
+
+    class OneHot(ObservationPreprocessor):
+        # Converts int observations (Discrete) into one-hot tensors (Box).
+        def recompute_output_observation_space(self, in_obs_space, in_act_space):
+            # Based on the input observation space (), return the output observation
+            # space. Implementing this method is crucial for the pipeline to know its output
+            # spaces, which are an important piece of information to construct the succeeding
+            # RLModule.
+            return gym.spaces.Box(0.0, 1.0, (in_obs_space.n,), np.float32)
+
+        def preprocess(self, observation):
+            # Convert an input observation (int) into a one-hot (float) tensor.
+            # Note that 99% of all connectors in RLlib operator in the "numpy space".
+            new_obs = np.zeros(shape=self.observation_space.shape, dtype=np.float32)
+            new_obs[observation] = 1.0
+            return new_obs
+
+
+Now that the preprocessor is written, you can plug it into your :py:class:`~ray.rllib.algorithms.algorithm_config.AlgorithmConfig`
+object and run an experiment with a Discrete observation env (observations are integers), for
+example `FrozenLake-v1 <https://gymnasium.farama.org/environments/toy_text/frozen_lake/>`__.
+
+
+.. testcode::
+
+    from ray.rllib.algorithms.ppo import PPOConfig
+
+    config = (
+        PPOConfig()
+
+        # Configure a simple 2x2 grid-world.
+        # ____
+        # |S |  <- S=start position
+        # | G|  <- G=goal position
+        # ----
+        .environment("FrozenLake-v1", env_config={"desc": ["SF", "FG"]})
+
+        # Plug our custom connector piece into the env-to-module pipeline.
+        .env_runners(env_to_module_connector=lambda env: OneHot())
+    )
+    algo = config.build()
+    # Train one iteration.
+    print(algo.train())
+
+
+ConnectorV2 pipelines in more detail
+------------------------------------
+
+Before diving into more complex and more powerful customization options of the different
+pipeline types (and walking through more examples), the generic inner workings of a
+ConnectorV2 pipeline should be explained in more detail.
```

Comment:
Passive voice. Just tell me what the section describes, don't tell me I need to learn this before learning more. Organize the content so that it makes sense in context and/or provide users with sufficient prompts and entry points to discover the content relevant to them.